spot-client/src/common/remote-control/xmpp-connection.js

```diff
@@ -657,21 +663,24 @@ export default class XmppConnection {
             type = 'join';
         }
 
-        return {
-            from,
-            localUpdate: from === this.getRoomFullJid(),
-            state: Array.from(presence.children)
-                .reduce((acc, child) => {
-                    let value = child.textContent;
+        const statusElement = presence.getElementsByTagNameNS('https://jitsi.org/spot', 'spot-status')[0];
+        let state;
 
-                    if (value === 'true' || value === 'false') {
-                        value = value === 'true';
-                    }
+        try {
+            state = statusElement ? JSON.parse(statusElement.textContent) : { };
+        } catch (error) {
+            logger.error('Failed to parse presence', {
+                from,
+                type
+            });
 
-                    acc[child.tagName] = value;
+            return undefined;
+        }
```

Comment:
The weird part will be if there is a time when people try to connect a new remote control to a TV running the old code. I wonder if we need to include some transitionary code to account for that edge case.

Reply:
yeah.. good point